portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.util.*; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes all comments. */
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+  private static final String COMMENT = "Comment";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(COMMENT);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
```

Comment:
This be replaced with datastore.delete(results.stream().map(entity->entity.getKey()).collect(toList())?

Please take a look at compiling the keys into an Iterable and using https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete-java.lang.Iterable-.

Ideally we should use a transaction I feel so that we can rollback if there was an issue deleting one key in the database. As this is written it is possible to continue deleting and not be sure if we deleted all or not, using a transaction will help with that. Perhaps the .delete(Iterable<>keys) would help with that I'm not sure. 
https://www.justsoftwaresolutions.co.uk/database/database_tip_use_transactions.html

Also take a look at Java lambdas and streams, this will help you in the future.
https://winterbe.com/posts/2014/07/31/java8-stream-tutorial-examples/
https://www.infoworld.com/article/3452018/get-started-with-lambda-expressions.html

Reply:
Added as a TODO